lib/next_ls/extensions/credo_extension/code_action/remove_debugger.ex

```diff
@@ -0,0 +1,122 @@
+defmodule NextLS.CredoExtension.CodeAction.RemoveDebugger do
+  @moduledoc false
+
+  alias GenLSP.Structures.CodeAction
+  alias GenLSP.Structures.Diagnostic
+  alias GenLSP.Structures.Position
+  alias GenLSP.Structures.Range
+  alias GenLSP.Structures.TextEdit
+  alias GenLSP.Structures.WorkspaceEdit
+  alias NextLS.EditHelpers
+  alias Sourceror.Zipper, as: Z
+
+  def new(diagnostic, text, uri) do
+    %Diagnostic{} = diagnostic
+    start = diagnostic.range.start
+
+    with {:ok, ast} <- parse(text),
+         {:ok, debugger_node} <- find_debugger(ast, start) do
+      indent = EditHelpers.get_indent(text, diagnostic.range.start.line)
+      ast_without_debugger = remove_debugger(debugger_node)
+      range = make_range(debugger_node)
+
+      formatted = Macro.to_string(ast_without_debugger)
+
+      [
+        %CodeAction{
+          title: make_title(debugger_node),
+          diagnostics: [diagnostic],
+          edit: %WorkspaceEdit{
+            changes: %{
+              uri => [
+                %TextEdit{
+                  new_text: EditHelpers.add_indent_to_edit(formatted, indent),
+                  range: range
+                }
+              ]
+            }
+          }
+        }
+      ]
+    else
+      _ ->
+        []
+    end
+  end
+
+  defp find_debugger(ast, position) do
+    pos = [line: position.line + 1, column: position.character + 1]
+
+    result =
+      ast
+      |> Z.zip()
+      |> Z.traverse(nil, fn tree, acc ->
+        node = Z.node(tree)
+        range = Sourceror.get_range(node)
+
+        if !acc &&
+             (matches_debug?(node, pos) || matches_pipe?(node, pos)) &&
+             Sourceror.compare_positions(range.start, pos) in [:lt, :eq] &&
+             Sourceror.compare_positions(range.end, pos) in [:gt, :eq] do
+          {tree, node}
+        else
+          {tree, acc}
+        end
```

Comment:
last thing i think.

i've patched the bug where IO.inspect and dbg warnings didn't have the correct columns (it previously backfilled column by just doing a string find on the line, i changed it to use the actual column metadata in the ast).

so these lines need to also account for column, not just line.

i'll get my PR up for credo and then you can test against that branch to confirm

Reply:
https://github.com/rrrene/credo/pull/1126

Reply:
looks like we need to patch the Mix.env, IO.puts, and IEx.pry rules as well. i can do it later unless you get to it first

but we can merge this once the above comment is merged, theres a chance credo doesn't merge and release these patches for a while

Reply:
I think this is fixed now